bin/propolis-server/src/lib/stats/virtual_machine.rs

```diff
@@ -0,0 +1,530 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2024 Oxide Computer Company
+
+//! Types for tracking statistics about virtual machine instances.
+
+// Propolis is built in a variety of configurations, including checks and tests
+// run on non-illumos machines where kstats are meaningless. This is a big
+// hammer, but a large number of the values in this module are not referenced in
+// those configurations, and so this is more straightfoward than littering the
+// code with cfg directives.
+#![cfg_attr(any(test, not(target_os = "illumos")), allow(dead_code))]
+
+use chrono::DateTime;
+use chrono::Utc;
+use oximeter::types::Cumulative;
+use oximeter::FieldType;
+use oximeter::FieldValue;
+use oximeter::Metric;
+use oximeter::Sample;
+use oximeter::Target;
+use std::collections::BTreeMap;
+use uuid::Uuid;
```

Comment:
style nit: please collapse down these includes

Reply:
Done in d61fd69